pkg/cli/zip.go

```diff
@@ -377,6 +381,71 @@ func maybeAddProfileSuffix(name string) string {
 	return name
 }
 
+type jobTrace struct {
+	jobID   jobspb.JobID
+	traceID int64
+}
+
+// dumpTraceableJobTraces collects the traces for some "traceable" jobs that are
+// in a running state. The job types in this list are the ones that have
+// explicitly implemented the TraceableJob interface.
+func (zc *debugZipContext) dumpTraceableJobTraces() error {
+	ctx := context.Background()
+	rows, err := zc.firstNodeSQLConn.Query(ctx,
+		`SELECT job_id, trace_id FROM crdb_internal.jobs WHERE job_type IN ($1, $2, $3, $4) AND status IN ($5, $6)`,
```

Comment:
I wonder if we want to write a custom query for this rather than something that uses the virtual table?  Namely, we have job_type and job_status in the system.jobs table so my guess is we can use that to construct a much better join with the job_info table than the one we'll get with this query.

Reply:
These types being listed out here kinda kills me; this list seems assured to rot when we add more tracable jobs and forget to update it.

I won't stand in the way of this PR since the list is correct right now, but can I ask we file an issue to come do something better here?

I wonder if we should long-term be adding a column to the jobs table that just says to run with verbose tracing yes/no, populated by the resumers what know their code has relevant trace events in it or via explicit action/settings.

Shorter-term maybe we hardcode the list somewhere in jobs or jobsprofiler and change the resumers to consult the list?

Again though, this can go in an issue rather than be changed before merging.

Reply:
We _already_ read every single row out of `crdb_internal.jobs` 90 lines above this during `queryAndDumpTables`, including explicitly selecting job_id, job_type and trace_id, and I briefly pondered if there was a way to pass something like map[string]interface{} to the dumper and then have an optional func on each table config in the registry that gets to see each row as it is dumped and edit the map or something so we could accumulate the job ids that are traceable as we go.

but it looks like we do the to string serializing into an io.Wrtier pretty deep in generic, non-zip specific sql code so maybe this isn't worth it to avoid going back to the jobs tables a second time.

Reply:
If nothing else: We could move this list to a package level var. Then we could write a test that iterates through all the job types in jobspb, grabs their resume, checks if they are a traceable job, and makes sure they are in the package level var.

Reply:
> grabs their resume
I thought about this, but we don't currently have all the ccl-side resumers in pkg/cli_test, which is why I suggested moving the list over to the jobs packages.

Again though, I'm _fine_ with the list as it is inline for right now as long as we file a follow-up.

Reply:
i agree with everything said here, this list is just bad news. I think the most tenable solution is to think about adding some more per-job state around observability. We've discussed wanting a "is being profiled" bit and now a way to know whether a job is tracing or not. Moving this list to jobs/jobprofiler makes sense but to write a test that can iterate over all the types, we'd have to import all the packages that have `RegisterConstructor` in their `init` methods. That also sounds painful and so I'd rather we just solve this by adding an info row or a column in the next release. I will file an issue and put this as a comment above this code.

@stevendanna re: rewriting this query, i will play with some query plans to see what works best.